sw/device/lib/dif/dif_rstmgr.h

```diff
@@ -0,0 +1,223 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RSTMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RSTMGR_H_
+
+/**
+ * @file
+ * @brief <a href="">Reset Manager</a> Device Interface Functions
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Reset Manager peripheral software reset control.
+ */
+typedef enum dif_rstmgr_software_reset {
+  /**
+   * Simple reset (release straight away).
+   */
+  kDifRstmgrSoftwareReset = 0,
+  kDifRstmgrSoftwareResetHold,    /**< Hold peripheral in reset. */
+  kDifRstmgrSoftwareResetRelease, /**< Release peripheral from reset. */
+} dif_rstmgr_software_reset_t;
+
+/**
+ * Reset Manager bitfield encoding the reset causes information.
+ */
+typedef uint32_t dif_rstmgr_causes_bitfield_t;
+
+/**
+ * Reset Manager possible reset causes enumeration.
+ *
+ * Invariants are used to extract information encoded in
+ * `dif_rstmgr_causes_bitfield_t`, which means that the values must correspond
+ * to the individual bits (0x1, 0x2, 0x4, ..., 0x80000000).
+ *
+ * Note: these must match the autogenerated register definitions.
+ *
+ * TODO: no documentation currently available, would be useful
+ *       for writing a more detailed description for the individual causes.
+ */
+typedef enum dif_rstmgr_cause {
+  kDifRstmgrCausePowerOn = 0x1,
+  kDifRstmgrCauseLowPowerExit = (0x1 << 1),
+  kDifRstmgrCauseWatchdog = (0x1 << 2),
+  kDifRstmgrCauseSecurityEscalation = (0x1 << 3),
+  kDifRstmgrCauseNonDebugModuleRequest = (0x1 << 4),
+} dif_rstmgr_cause_t;
+
+/**
+ * Reset Manager software reset available peripherals.
+ */
+typedef enum dif_rstmgr_peripheral {
+  kDifRstmgrPeripheralSpi = 0,
+  kDifRstmgrPeripheralUsb,
+  kDifRstmgrPeripheralLast = kDifRstmgrPeripheralUsb,
```

Comment:
These are missing comments (and the last one should be marked \internal).

Reply:
Done.